internal/integritymonitor/integritymonitor.go

```diff
@@ -0,0 +1,266 @@
+package integritymonitor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/core/models"
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/core/services"
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/repositories"
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/repositories/data"
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/utils/process"
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/walker"
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/worker"
+	"github.com/ScienceSoft-Inc/integrity-sum/pkg/alerts"
+	"github.com/ScienceSoft-Inc/integrity-sum/pkg/api"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+const (
+	IntegrityMessageNewFileFound = "new file found"
+	IntegrityMessageFileDeleted  = "file deleted"
+	IntegrityMessageFileMismatch = "file content mismatch"
+)
+
+func GetProcessPath(procName string, path string) (string, error) {
+	pid, err := process.GetPID(procName)
+	if err != nil {
+		return "", fmt.Errorf("failed build process path: %w", err)
+	}
+	return fmt.Sprintf("/proc/%d/root/%s", pid, path), nil
+}
+
+func SetupIntegrity(ctx context.Context, monitoringDirectory string, log *logrus.Logger, deploymentData *models.DeploymentData) error {
+	log.Debug("begin setup integrity")
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	errC := make(chan error)
+	defer close(errC)
+
+	log.Trace("calculate & save hashes...")
+	select {
+	case <-ctx.Done():
+		log.Error(ctx.Err())
+		return ctx.Err()
+
+	case countHashes := <-saveHashes(
+		ctx,
+		worker.WorkersPool(
+			viper.GetInt("count-workers"),
+			walker.ChanWalkDir(ctx, monitoringDirectory, log),
+			worker.NewWorker(ctx, viper.GetString("algorithm"), log),
+		),
+		deploymentData,
+		errC,
+	):
+		log.WithField("countHashes", countHashes).Info("hashes stored successfully")
+		log.Debug("end setup integrity")
+		return nil
+
+	case err := <-errC:
+		log.WithError(err).Error("setup integrity failed")
+		return err
+	}
+}
+
+func CheckIntegrity(ctx context.Context, log *logrus.Logger, monitoringDirectory string, alertSender alerts.Sender, deploymentData *models.DeploymentData, kubeClient *services.KubeClient) error {
+	log.Debug("begin check integrity")
```

Comment:
The `alertSender alerts.Sender` and `kubeClient *services.KubeClient` actually like a const in our case. Once inited they will not change. So, we may use the DB flow instead: make global variables and use them directly.

Up to you.

Reply:
lets do this in another PR